ballerina/utils.bal

```diff
@@ -15,21 +15,146 @@
 // under the License.
 
 import ballerina/jballerina.java;
+import ballerina/time;
+import ballerina/regex;
 
-isolated function getRandomUUID() returns handle = @java:Method {
-    name: "randomUUID",
-    'class: "java.util.UUID"
-} external;
+isolated function generateLeastSigBits() returns int {
+    int random63BitLong = nextLong() & 0x3FFFFFFFFFFFFFFF;
+    int variant3BitFlag = -0x8000000000000000;
+    return random63BitLong + variant3BitFlag;
+}
 
-isolated function getBytesFromUUID(string uuid) returns byte[] = @java:Method {
-    name: "getBytesFromUUID",
-    'class: "io.ballerina.stdlib.uuid.nativeimpl.Util"
-} external;
+isolated function generateMostSigBits() returns int {
+    time:Utc duration = time:utcAddSeconds(time:utcNow(), GREGORIAN_TIME_IN_SECONDS);
+    int timeInNanos = duration[0] * 10000000 + <int>duration[1] * 100;
 
-isolated function getUUIDFromBytes(byte[] uuid) returns string = @java:Method {
-    name: "getUUIDFromBytes",
-    'class: "io.ballerina.stdlib.uuid.nativeimpl.Util"
-} external;
+    int leastSigBitOfTime = (timeInNanos & 0x000000000000FFFF) >> 4;
+    int uuidVersion = 1 << 12;
+    return (timeInNanos & 0x7FFFFFFFFFFF0000) + uuidVersion + leastSigBitOfTime;
+}
+
+isolated function getBytesFromUuid(string uuid) returns byte[]|Error {
+    Uuid uuidRecord = check toRecord(uuid);
+
+    int msb = check getMostSignificantBits(uuidRecord);
+    int lsb = check getLeastSignificantBits(uuid, uuidRecord);
+    return bitsToBytes(msb, lsb);
+}
+
+isolated function getMostSignificantBits(Uuid uuid) returns int|Error {
```

Comment:
```suggestion
isolated function getMostSigBits(Uuid uuid) returns int|Error {
```

Reply:
Consistent with other apis.

Reply:
Fixed in [760efaa](https://github.com/ballerina-platform/module-ballerina-uuid/pull/376/commits/760efaaf5200a5a24322a76289627d36330a100a)